compiler/backend/src/org/jetbrains/kotlin/codegen/inline/MethodInliner.kt

```diff
@@ -825,7 +823,7 @@ class MethodInliner(
                 getFunctionalArgumentIfExists((insnNode as VarInsnNode).`var`)
             insnNode is FieldInsnNode && insnNode.name.startsWith(CAPTURED_FIELD_FOLD_PREFIX) ->
                 findCapturedField(insnNode, nodeRemapper).functionalArgument
-            insnNode is FieldInsnNode && insnNode.isSuspendLambdaCapturedByOuterObjectOrLambda(inliningContext) ->
+            insnNode is FieldInsnNode && inliningContext.root.sourceCompilerForInline.isSuspendLambdaCapturedByOuterObjectOrLambda(insnNode.name) ->
```

Comment:
Could some `isSuspendLambdaCapturedByOuterObjectOrLambda` cases arise from mixed compilation (old capturing scheme vs new one with explicit parameters propagation)?  Should it be some more tricky in ir implementation than:
```
    override fun isSuspendLambdaCapturedByOuterObjectOrLambda(name: String): Boolean =
        false // IR does not capture variables through outer this
``` ?

Could you check do we have tests for such cases?

Reply:
In this context "outer" means in relation to the inline function call, e.g.
```
inline fun something(crossinline x: suspend () -> Unit) {
    val y: suspend () -> Unit = { // <- outer object
        inlineFunction { x() } // <- call of a suspend lambda captured by said outer object
    }
    y()
}
```
so there can be no mixed compilation here - the referenced value is in the source file that is currently being compiled.